activerecord/lib/active_record/model_schema.rb

```diff
@@ -484,7 +525,11 @@ def load_schema
         end
 
         def load_schema!
-          @columns_hash = connection.schema_cache.columns_hash(table_name).except(*ignored_columns)
+          @columns_hash = connection.schema_cache.columns_hash(table_name)
+
+          @columns_hash = @columns_hash.except(*ignored_columns)
+          @columns_hash = @columns_hash.slice(primary_key, *attributes_to_define_after_schema_loads.keys) unless define_attributes_from_schema
```

Comment:
I still think we should not be modifying the `columns_hash`. The model columns should be still present on the `columns` call.

Reply:
Do you think i should make it so ignored_columns also doesn't modify the hash then?